docs/rfcs/0003-ecommerce-models.md

```diff
@@ -0,0 +1,136 @@
+## Title for RFC
+
+ecommerce workflow
+
+#### Abstract
+
+This describes the various states of ecommerce models and where and when we can and cannot
+modify data.
+
+#### Product
+
+A `Product` has a `price`, `description`, and a `GenericForeignKey` `content_object` which can
+refer to any Django model instance. An admin user should be able to edit
+`description` or `price` without affecting any previous purchase, so we need to also store
+this separately when the purchase is made.
+
+#### Basket
+
+A `Basket` has foreign keys to an `Order` and a `User`. The `User` foreign key must always exist.
+
+A `Basket` may be editable or not editable, depending on the `editable` boolean field. In general
+a `Basket` starts off editable so that the user can add products or a coupon. Then it becomes not
+editable once it is passed to CyberSource for payment.
+
+When a `Basket` is editable it should not have a link to an `Order`, and when it is not editable
+it should have a link to an `Order`. This makes the `editable` flag a little redundant but this
+makes the intent more explicit, and it provides flexibility in the future if we want to change
+the logic around when `Order` is created and linked to a `Basket`.
+
+There should be at most one `Basket` for a user which is editable.
+This is the user's active basket. If there is no editable `Basket` for the user,
+it should be created once the user adds a product or a coupon.
```

Comment:
should a basket expire after a certain point, provided that a user hasn't started the check-out flow?

Reply:
That's a good question, I'm not sure when the basket should be cleared in general. I'm not sure it will be useful to expire in this instance since there will only be a course or a program in the basket at once, and we can clear the basket in the view.